components/mediators/datamapper/org.wso2.carbon.mediator.datamapper.engine/src/main/java/org/wso2/carbon/mediator/datamapper/engine/input/readers/XMLInputReader.java

```diff
@@ -200,6 +200,9 @@ public String xmlTraverse(OMElement omElement, String prevElementName, Map jsonS
                 elementType = getObjectTextElementType(jsonSchemaMap, nameSpaceLocalName);
             }
         }
+        /** If an object/element(String/boolean/integer/number) property contains xis:nil=true
+         * need  to avoid writing those fields*/
+        if (!isXsiNil(omElement)) {
         /* If there is text in the OMElement */
         if (DataMapperEngineConstants.STRING_ELEMENT_TYPE.equals(elementType)
```

Comment:
Fixed with https://github.com/wso2/carbon-mediation/pull/808/commits/b6a78dc1378e9099374962888a02147568f2dce2 and https://github.com/wso2/carbon-mediation/pull/808/commits/8d24acfd385c1e1a7417122f5f8fb755fda2f461